File: SmartPay/ClientApp/src/api/models/Product.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { CheckProduct } from './CheckProduct';
import type { Merchant } from './Merchant';
import type { SubCategory } from './SubCategory';

export type Product = {
    id?: number;
    name?: string | null;
    price?: number;
    checks?: Array<CheckProduct> | null;
    merchantId?: number;
    merchant?: Merchant;
    categoryId?: number;
    category?: SubCategory;
};

File: SmartPay/Models/Product.cs
namespace SmartPay.Models;

public class Product
{
    public int Id { get; set; }
    
    public string Name { get; set; }
    public int Price { get; set; }
    
    public List<CheckProduct> Checks { get; set; }
    
    public int MerchantId { get; set; }
    public Merchant Merchant { get; set; }
    public int CategoryId { get; set; }
    public SubCategory? Category { get; set; }
}

public class ProductViewModel
{
    public int Id { get; set; }
    
    public string Name { get; set; }
    public int Price { get; set; }
    
    public MerchantViewModel Merchant { get; set; }
    public SubCategoryProductViewModel? Category { get; set; }
}